Szafirmag/Zadanie 3/Zadanie3.cc

```diff
@@ -0,0 +1,83 @@
+#include <algorithm>
+#include <cassert>
+#include <chrono>
+#include <functional>
+#include <iostream>
+#include <thread>
+#include <numeric>
+#include <vector>
+
+#define assertm(exp, msg) assert(((void)msg, exp))
+
+using namespace std;
+
+constexpr size_t minimumSize = 25000;
+
+decltype(chrono::steady_clock::now()) start3;
+decltype(start3) stop3;
+decltype(start3) start4;
+decltype(start4) stop4;
+
+template <typename IT, typename V>
+int pararelCountIF(IT first, IT last, V value) {
```

Comment:
parametr `V value` lepiej nazwać `P predicate`. Dzięki szablonom zadziała to na dowolnym typie, a sama nazwa predykat jest ogólniejsza, więc jest bardziej intuicyjne że można tam lambdę przekazać.